typing/global_module.ml

```diff
@@ -307,3 +308,86 @@ let has_arguments t =
   match t with
   | { head = _; visible_args = []; hidden_args = [] } -> false
   | _ -> true
+
+module Precision = struct
+  type t = Exact | Approximate
+
+  let print ppf = function
+    | Exact -> Format.fprintf ppf "exact"
+    | Approximate -> Format.fprintf ppf "approx"
+
+  let output = Misc.output_of_print print
+
+  let equal t1 t2 =
+    match t1, t2 with
+    | Exact, Exact
+    | Approximate, Approximate -> true
+    | (Exact | Approximate), _ -> false
+end
```

Comment:
I wonder if this date type is needed. It seems currently it's only used to trigger internal errors? (Which I'm fine with. Just want to point out that this is not used for anthing that's visible to the users.)